### ISSUE TITLE
False positive on lambda variables

### ISSUE THREAD

**Issue by johnjaylward** [checkstyle/checkstyle#2895]
Using the default version of checkstyle available in gradle 2.10, an expression like:
```java
public static BigDecimal sum(final Collection<? extends HasAmount> amounts) {
    return amounts.parallelStream().filter(i -> i != null).map(a -> a.getAmount())
            .reduce((x, y) -> x.add(y)).orElse(BigDecimal.ZERO);
}
```

will yield 2 errors:
* FinalLocalVariableCheck, Priority: High<br /> Variable 'x' should be declared final.
* FinalLocalVariableCheck, Priority: High<br /> Variable 'y' should be declared final.

It is syntactically incorrect to add the final keyword to either x or y in the lambda definition. The code would not compile.

**Comment by romani** [comment]
please report issue as described there http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by johnjaylward** [comment]
Looks like the latest version fixes the issue. Using it on the command line and specifying the toolVersion in gradle removes the false positive. Thank you.

**Comment by romani** [comment]
we are releasing fixes each end of month, subscribe you our news feed for notifications.
